Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,847 @@
+                @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                @{N = 'ResourceName'; E = { $_.Name } }, 
+                @{N = 'TLSVersion'; E = { $_.$([Constants]::ParameterName_TLS) } },
+                @{N = 'SSLEnforcement'; E = { $_.sslEnforcement } }
+            }
+            catch {
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Warning)	
+                Write-Host "Skipping the Resource:  [$($_.ResourceName)]..." -ForegroundColor $([Constants]::MessageType.Warning)	
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                $logResource.Add("ResourceName", ($_.ResourceName))	
+                $logResource.Add("Reason", "Valid resource id(s) not found in input json file.")    	
+                $logSkippedResources += $logResource	
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Error fetching Azure Database for PostgreSQL single server(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }	
+        }	
+    }
+    else {	
+        if ([String]::IsNullOrWhiteSpace($FilePath)) {
+            Write-Host "Fetching all Azure Database for PostgreSQL single server(s) in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+
+            # Get all Azure Database for PostgreSQL single server(s) in a Subscription
+            $servers = @();
+            $servers = Get-AzPostgreSqlServer -ErrorAction Stop
+            $servers | ForEach-Object {
+                $DBForPostgreSQLSingleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                @{N = 'ResourceName'; E = { $_.Name } }, 
+                @{N = 'TLSVersion'; E = { $_.$([Constants]::ParameterName_TLS) } },
+                @{N = 'SSLEnforcement'; E = { $_.$([Constants]::ParameterName_SSL) } }
+            }        
+        }
+        else {
+            if (-not (Test-Path -Path $FilePath)) {
+                Write-Host "ERROR: Input file: [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+
+            Write-Host "Fetching all Azure Database for PostgreSQL single server(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $DBForPostgreSQLResources = Import-Csv -LiteralPath $FilePath
+
+            $validDBForPostgreSQLSSResources = $DBForPostgreSQLResources | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
```

Comment:
Here we are not checking if validDBForPostgreSQLSSResources is empty then?

Reply:
checking in the further code as explained.